### PR TITLE
feat: User provisioning & Group sync

### PR DESCRIPTION
**Pull request by Pranay105**
## Checklist
I have:
- [x] run the content through Grammarly
- [ ] linked to sample apps when relevant
- [x] added the meta description for each page in the PR
- [ ] minimized the callouts and added only when necessary
- [x] added the `queryString` parameter to the Tabs (if used)
- [ ] masked PII in images. For example, login credentials, account details, and more
- [ ] added images only when necessary
- [ ] deleted the images that are no longer used for the updated pages in the PR
- [ ] followed the image file naming convention while renaming or adding new images. (Use lowercase letters, dashes between words, and be as descriptive as possible)
- [ ] used the `<figure/>` tag instead of a markdown representation for images
- [ ] added the `<figcaption/>` tag to add a caption to the image
- [ ] added the `alt` attribute in the `` tag
- [x] verified and updated the cross-references or created redirect rules for the changed or removed content 
- [x] reviewed and applied the style changes for UI formatting. For example, Bold the UI elements(Buttons on screen) used in the doc.

### CODE REVIEW

File: website/docs/advanced-concepts/user-provisioning-group-sync.mdx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import TabItem from '@theme/TabItem';
+
+# Setup SCIM Provisioning
+
+:::info
+SCIM Provisioning is only available in Appsmith's [Enterprise Edition](https://www.appsmith.com/pricing).
+:::
+
+
+This page shows how to set up SCIM on your Appsmith instance to streamline the user lifecycle management based on your SSO provider.
```
Comment (danciaclara) [inline website/docs/advanced-concepts/user-provisioning-group-sync.mdx:15]:
```suggestion
This page shows how to configure Appsmith for SCIM to enable the automatic provisioning of users and groups based on your SSO provider.
```

File: website/docs/advanced-concepts/user-provisioning-group-sync.mdx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+
+## Prerequisites
+
+- A self-hosted Appsmith instance. See the [installation guides](/getting-started/setup/installation-guides) for deploying Appsmith.
+- A SAML SSO provider configured on your Appsmith instance.
```
Comment (danciaclara) [inline website/docs/advanced-concepts/user-provisioning-group-sync.mdx:21]:
```suggestion
- A SAML SSO provider configured on your Appsmith instance. See [SAML SSO](/getting-started/setup/instance-configuration/authentication/security-assertion-markup-language-saml) for instructions.
```

File: website/docs/advanced-concepts/user-provisioning-group-sync.mdx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+## Enable SCIM provisioning on Appsmith
+
+1. On your instance, go to **Admin Settings > Access Control** and click **Provisioning**.
+2. On the **User Provisioning & Group Sync** page, click **Configure**.
+3. Copy the **SCIM API endpoint** URL to add it later in your IdP application.
+4. Under **API key to setup SCIM**, click the **Generate API key** button and copy the API key to add it later in your IdP application.
+
```
Comment (danciaclara) [inline website/docs/advanced-concepts/user-provisioning-group-sync.mdx:29]: add screenshot on the setup page on Appsmith

**Review by danciaclara — CHANGES_REQUESTED** [review]
In both AD and Okta at the bottom, you need to add

## See Also
- Setup SCIM provisioning


also are not adding any screenshots for both AD and Okta to help users to align in case there are many back and forth steps happening.

File: website/docs/advanced-concepts/user-provisioning-group-sync.mdx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+## Configure SSO provider
+
+To complete the setup, go to your IdP application that is connected to Appsmith and follow the steps below:
+
+<Tabs queryString="scim">
+<TabItem label="Active Direcotry (SAML)" value="scim-ad-saml">
```
Comment (danciaclara) [inline website/docs/advanced-concepts/user-provisioning-group-sync.mdx:42]:
```suggestion
<TabItem label="Active Directory (SAML)" value="scim-ad-saml">
```

Comment (danciaclara) [reply website/docs/advanced-concepts/user-provisioning-group-sync.mdx:42]: I think its misspelt in other places also

File: website/docs/advanced-concepts/user-provisioning-group-sync.mdx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+3. On the **Provisioning** homepage, click **Provisioning** on the sidebar and follow the steps below:
+
+    a. Select **Automatic** from the **Provision Mode** dropdown.
+
+    b. In the **Tenant URL** field, paste the **SCIM API endpoint** URL that you copied in step 3 under [Enable SCIM provisioning](/advanced-concepts/user-provisioning-group-sync#enable-scim-provisioning-on-appsmith) section.
+
+    c. In the **Secret Token** field, paste the **API key** that you copied in step 4 under [Enable SCIM provisioning](/advanced-concepts/user-provisioning-group-sync#enable-scim-provisioning-on-appsmith) section.
+
+    d. Click the **Test Connection** button to check if the added credentials are authorized to enable provisioning.
+
```
Comment (danciaclara) [inline website/docs/advanced-concepts/user-provisioning-group-sync.mdx:57]: This section needs to have a separate heading

**Assign users and groups to the application**

**Comment by Pranay105** [comment]
> In both AD and Okta at the bottom, you need to add
> 
> ## See Also
> * Setup SCIM provisioning
> 
> also are we not adding any screenshots for both AD and Okta to help users to align in case there are many back and forth steps happening?

I did not feel the need for screenshots as such. During Attribute Mapping steps, there is a possibility of a little confusion, but I still feel the instructions are enough.

File: website/docs/advanced-concepts/user-provisioning-group-sync.mdx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    i. Go back to the **Provisioning** page, turn the toggle **On** for **Provisioning status** present at the bottom. 
+
+4. Go to **Users and Groups** from the sidebar, and click **Add user/group** to add a user or group to your application. These users or groups get automatically synced and added to your Appsmith instance.
+
+5. If you want to sync a user or group manually, go to the **Provision on Demand** from the sidebar, and enter a user or group name in the **Select a user or group** field. 
+
+</TabItem>
+
+<TabItem label="Okta (SAML)" value="scim-okta-saml">
+
```
Comment (danciaclara) [inline website/docs/advanced-concepts/user-provisioning-group-sync.mdx:82]:
```suggestion
### Add SCIM provisioning
```

File: website/docs/advanced-concepts/user-provisioning-group-sync.mdx [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+1. On the Okta homepage, go to **Applications > Applications** on the sidebar, and open the SAML application that is already connected to Appsmith.
+2. On the application page, go to the **General** tab, click **Edit** on **App Settings** and select the **Enable SCIM provisioning** checkbox. Click **Save**.
+3. Go to the **Provisioning** tab, click **Edit** on SCIM connection.
```
Comment (danciaclara) [inline website/docs/advanced-concepts/user-provisioning-group-sync.mdx:85]: Above this add, 

### Choose provisioning options

File: website/docs/advanced-concepts/user-provisioning-group-sync.mdx [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+4. Under **API key to setup SCIM**, click the **Generate API key** button and copy the API key to add it later in your IdP application.
+
+
+## Configure SSO provider
+
+To complete the setup, go to your IdP application that is connected to Appsmith and follow the steps below:
+
+<Tabs queryString="scim">
+<TabItem label="Active Direcotry (SAML)" value="scim-ad-saml">
+
```
Comment (danciaclara) [inline website/docs/advanced-concepts/user-provisioning-group-sync.mdx:43]:
```suggestion
### Configure the enterprise application
```

**Review by danciaclara — CHANGES_REQUESTED** [review]

**Review by danciaclara — DISMISSED** [review]

**Review by danciaclara — COMMENTED** [review]
Hey I'll approve it once Akhil completes tech review

File: website/docs/advanced-concepts/user-provisioning-group-sync.mdx [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+    c. In the **Secret Token** field, paste the **API key** that you copied in step 4 under [Enable SCIM provisioning](/advanced-concepts/user-provisioning-group-sync#enable-scim-provisioning-on-appsmith) section.
+
+    d. Click the **Test Connection** button to check if the added credentials are authorized to enable provisioning.
+
+    e. Expand the **Mappings** section, click **Provision Azure Active Directory Users**, and scroll down to **Attribute Mappings** to configure attributes for users. Delete all the attributes except the following:
+
+    - **userPrincipalName**
+    - **displayName**
+    - **Switch([IsSoftDeleted], , "False", "True", "True", "False")**
```
Comment (Pranay105) [inline website/docs/advanced-concepts/user-provisioning-group-sync.mdx:62]: Ignore the Linter error for this. This is a UI element in AD that needs to be mentioned as it is in the product

**Review by infinitetrooper — DISMISSED** [review]

File: website/docs/advanced-concepts/user-provisioning-group-sync.mdx [hunk 10]
```diff
@@ -34,7 +34,7 @@ This page shows how to configure Appsmith for SCIM to enable the automatic provi
 
 
 
-## Configure SSO provider
+## Configure Identity provider
```
Comment (infinitetrooper) [inline website/docs/advanced-concepts/user-provisioning-group-sync.mdx:37]: I in Identity should be lowercase right?

**Review by danciaclara — APPROVED** [review]

**Pull request merged**